kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -112,10 +113,12 @@ termUnification notSimplifier = \term1 term2 ->
         TermLike RewritingVariableName ->
         unifier (Pattern RewritingVariableName)
     termUnificationWorker pat1 pat2 = do
+        injSimplifier <- Simplifier.askInjSimplifier
+        OverloadSimplifier{isOverloaded} <- Simplifier.askOverloadSimplifier
         let maybeTermUnification ::
                 MaybeT unifier (Pattern RewritingVariableName)
             maybeTermUnification =
-                maybeTermAnd notSimplifier termUnificationWorker pat1 pat2
+                maybeTermAnd notSimplifier termUnificationWorker injSimplifier isOverloaded pat1 pat2
```

Comment:
Why are we passing the new parameters? `maybeTermAnd` can retrieve these from its `unifier` monad.

Reply:
They are used in some match functions (`matchSortInjectionAndEquals` for instance) so we do not have access at that point to the `unifier` monad.